### DISCUSSION TITLE
There should be a command to toggle high verbosity on and off in browse mode

### DISCUSSION THREAD

**Discussion by Gene703122** [nvaccess/nvda#14942] [Ideas]
### Is your feature request related to a problem? Please describe.
At present, an enormous amount of verbosity is spoken by default in Browse Mode.  This may be desirable for certain uses but it is often a real waste of time and impediment to using the Internet efficiently in a lot of cases.  

### Describe the solution you'd like
There should be a command to allow users to toggle most verbosity on and off.  At present, all settings must be changed manually and this is not a reasonable way for people to change what they hear when they are using the Internet in different ways and for different purposes.

### Describe alternatives you've considered
None.
### Additional context
Look at all the verbosity settings regarding browse mode in document presentation, heading level, landmarks, block quotes and the list goes on and on.  

If I am doing research, I may want to hear a lot of verbosity.  If I'm designing a web page, I may want to hear lots of verbosity.  If I'm looking down The New York Times home page, which is designed so that you can move from story headline to story headline by pressing h from the top of the page, each headline link is a heading, I really don't want my time wasted by hearing heading level x every single time I move by heading from story to story.  I want to immediately hear the text of the headline.

Also, in the punctuation/symbol pronunciation dialog, shape after shape is set to none.  These shapes are almost always nothing but eye candy and should be set to all when you use the toggle verbosity command.  I really don't need, and I got along perfectly well for more than two decades, without hearing hollow right pointing triangle, downward pointing filled triangle. right facing arrow, etc.  Screen-readers have started announcing enormous amounts of eye candy lately and it is not reasonable that these symbols should have to be changed individually in the dialog.  

When one first enters browse mode, a first run dialog could be presented describing the on/off toggle so people will know about it ant, what it does, and the command to use.  
or there could be a dialog describing the command and a learn more link where a more detailed discussion would be given.

However, this is done, it is something that is badly needed and will result in much more efficient and satisfactory use of the Internet for different uses.

**Comment by Jykke67** [comment]
As a long time user of JAWS, I'm also frustrated on NVDA's verbosity in browsing mode. In JAWS I've modified the announcements by modifying the default Sound Scheme by replacing heading level 1 with h1 etc. Instead of no information at all, I still get it but in a much more convenient way than by default.

But why not have this turned completely off like you've suggested? Pressing "h" would still move to next heading etc. Perhaps I would still need to know which level of heading is present just by hearing "h 1" etc.

But yes, the additional information that you've mentioned is just frustrating.

**Comment by burmancomp** [comment]
Maybe more is better here: on, custom messages and off. Then everyone can decide what he/she wants to hear/read.

**Comment by Jykke67** [comment]
Just for addition, could the default announcements also be abbreviated, or would just turning them on or off be better? Perhaps the latter, because modifying defaults would also require translation to every NVDA supported language.

**Comment by burmancomp** [comment]
There is a dictionary in source code (source\controlTypes\role.py) defining labelsfor control types. This seems to be modified in source\braille.py for braille labelling.

There is likely additional code affecting on how things are spoken or shown in braille but maybe somebody could tell what could be achieved if user could customize these dictionaries as a part of solution.

If user for example thinks "h 1" would be fine for "heading level 1", why could not he/she modify this?

**Comment by irrah68** [comment]
> Maybe more is better here: on, custom messages and off. Then everyone can decide what he/she wants to hear/read.

This sounds very good. "On" would mean NVDA's default values, "custom" would mean that an user can change the way how an element will be spoken and "off" would mean that nothing will be spoken: no links, no frames etc.

I want to hear heading levels, because they are useful when navigating a site and make all the things faster, but I don't want to hear heading levels said in a long form. I have set JAWS speak them as H, and now, when there is a Heading level 2, I'll hear H 2. I'd would be a good idea to add this kind of functionality also to NVDA.

**Comment by cary-rowen** [comment]
I think it's important to allow the user to change the order of roles of elements or objects.

**Comment by akash07k** [comment]
Related to #7232 and #11102

**Comment by Qchristensen** [comment]
One way to achieve what you want with custom verbosity levels is with configuration profiles:

1) Setup one (or multiple) configuration profiles, which are manually activated.
2) Assign a keystroke in NVDA's inpu gestures dialog
3) Load the chosen configuration profile.
4) Make any adjustments needed.
5) Press NVDA+control+c to save the configuration.

Repeat as needed.  While this may not be as convenient as an out of the box verbosity setting keystroke, it does have the advantage of being able to add other things to it - for instance, you might have a profile to quickly read just the key inforamtion - you can set what elements and other information is automatically read, you can adjust the punctuation level (so you don't hear commas and full stops etc) and you can speed the voice up.  You can have a proof reading profile which has a slower voice with more punctuation and information read.

Also, if we did have a pre-set "high verbosity" setting and "low verbosity" setting inbuilt - then guaranteed it would not be exactly how you want it (or how others want it) - so it would need to be customisable, and by that time, it wouldn't be much different to using a configuration profile, except potentially less flexible.

For customising how NVDA reads information like reading H1, H2, H3 instead of "heading level 1", etc.  That's a good example for what you want, and probably the main one?  If you press B to jump to a button, it's a button.  But if you move to a heading, it COULD be one of various levels, and it's probably the main one that many people already also know as H1, H2, H3 etc.  I'm not sure what you would shorten button, or edit box or nonLinked text to?  You could use first letters for some, but then is "c" a combo box, or a checkbox, and not everyone is going to know those as immediately.  But perhaps an issue for customising NVDA's inbuilt strings could be worthwhile - that way you could change it as you want, without needing to do it via a dictionary entry, which would also change any time that particular word is written anywhere.

**Reply by burmancomp** [reply]
> But perhaps an issue for customising NVDA's inbuilt strings could be worthwhile - that way you could change it as you want, without needing to do it via a dictionary entry, which would also change any time that particular word is written anywhere.

I think this could be a good approach.

**Reply by Gene703122** [reply]
You could use Profiles but how many users use or know about them?  In addition, you would need different profiles if you want more than two levels of verbosity because assumedly, you would be using the same browser most of the time and you would have to make each profile manually triggered.  I'm saying that, for a lot of users, what is needed is a way to allow changing verbosity settings in browse mode with simple commands, that might even be activated with short cuts.  For example, high verbosity, medium verbosity and low verbosity.  I didn't know this until very recently, after I created the ticket, but JAWS has just such options so people can quickly and conveniently change the amount of verbosity without having to change setting after setting after setting in document presentation and creating profiles.  

Someone might want a high amount of verbosity when researching a college paper, and low verbosity when reading for pleasure or interest on a news paper site.  The reader who reads a New York Times article for pleasure or as a general interested reader doesn't need to hear, list with five items or other such information.  They are listening as a talking book listener would.  No one wants or expects talking book readers to read like this.

**Comment by burmancomp** [comment]
Is this somewhat related also to issue #46?

**Comment by XLTechie** [comment]
This is exactly issue #46.
Which has more context plus age.

**Comment by Gene703122** [comment]
I believe a lot of users of any screen-reader don't know much about it.  They take things as they are in large part.  If we could ask them what they think of all the verbosity in Browse Mode, I think a lot of NVDA users would complain.  

I believe NVDA users who use this forum are a small group.  I think the small number of comments may well not reflect the unheard discontent of a lot of users who don't know where to complain, or even that there are alternatives that could be implemented.  

A lot of attention is placed on moving quickly through a site to find something.  No importance seems to be placed on the enormous amount of time wasted by a lot of users by hearing verbosity of no interest or use to them.  If you are reading for pleasure or information, as when reading a newspaper article or a help article discussing how you change a setting, most users don't benefit from hearing list of four items.  Excess verbosity, repeated over and over on page after page wastes an enormous amount of time.

You may want high verbosity when researching a college paper and low verbosity when looking through a newspaper site and reading articles.

I didn't know this when I filed the ticket, but I very recently learned that JAWS has just the kinds of settings I am talking about.  The user can set verbosity in The Virtual PC Cursor to high, medium, and low.  If those aren't the exact words, equivalent wording is used.

The point isn't that NVDA should do something just because JAWS does.  the point is that JAWS developers have met a need that I believe is important.

**Comment by XLTechie** [comment]
Would shipping NVDA with a few default verbosity profiles, triggered by 
defined key commands, solve this?
That is, assuming they were documented in the userguide?

**Comment by burmancomp** [comment]
There are setting dialog for punctuation/symbol pronunciation. I feel there is a lot of similarity with verbosity problem: what is said (compare with pronunciation) and when (compare with symbol level).

As to verbosity, user could define (if he/she is not happy with defaults) how given thing (heading, list end etc.) should be spoken (and brailled) and when (starting from given verbosity level) it should be spoken (brailled).

**Comment by hozosch** [comment]
Sorry, so this discussion was already present here. That last comment is exactly what I'm looking for as well in my issue I created yesterday.

**Comment by Jykke67** [comment]
No need to be sorry for this, because this issue is still present.

Like @irrah68 and some others have done, I have set JAWS to say h1 for heading level 1 etc. by setting verbosity level to low and by modifying sound schemes, which means that the changes does not effect elsewhere like when typing or reading text etc.

This is the way that this should be done also in NVDA like already explained.

**Comment by Adriani90** [comment]
My opinion on this, I am not in favor of pre-defining too much what high verbosity and low verbosity is. But i agree that users should have the flexibility to change the verbosity according to their needs.
I think the document formating settings is a very good start already. If you generally find heading level 1 too long, there should be a possibility to change this to a shorter reoprting style. But doing this website by website is not really efficient. I rather set my settings for the whole browser profile (e.g. Firefox) and that's it.
If I am good with h1 instead of heading level 1, then it doesn't matter whether it is on New York Times or Google search results or what ever.

**Comment by Jykke67** [comment]
Yes, browser spesific configuration profiles would be a step forward, but an opportunity to customize the verbosity in general settings would be an ideal solution.

I have told how this is done in JAWS by sound schemes. When editing a scheme, one is able to edit each announcement for a link, heading etc. separately, and they can be even muted.

This means that typed and read text is not affected, but just the announcements without any browser or other application dependency are. One is able to replace heading level x with anything that he/she wants. I have replaced them with h1 etc.

The modified announcements do override the defaults everywhere without affecting any typed or read text, because just the announcements are affected. But this is how it is done in JAWS.

But if no sound schemes, then defaults, customized and no announcements like @burmancomp and @irrah68 and some others have suggested? JAWS is JAWS and NVDA is NVDA.

**Comment by Qchristensen** [comment]
I think there are a few different points here.  If we can break them down into specific tasks, they can be created as issues.  For instance, from the original issue at the top of the thread:

1) Allow changing a bunch of (Document formatting?) settings at once.  This is already possible with configuration profiles, but would need to be setup by the user.  If we did have a "document verbosity" toggle which flicked most of those on or off, we would need to consider, how would that work when used in combination with configuration profiles?  Which would have priority?  And would having an extra feature like that cause more confusion?  We already have symbol / punctuation level etc, so would need to be clear about which option controlled what speech.

2) Many emoji and other symbols are set to verbosity level "none" and the request would be to set these to all.  That's a straightforward one.  Please create an issue with a list.

3) Ability to adjust inbuilt strings, for instance "Heading level 1" to "h1".   I can see the argument here, and it seems like the kind of thing you could do in speech dictionaries, and you can, but it takes a little trial and error.  Looking at the speech viewer, when I go to a heading level 1, NVDA reads: "heading    level 1" - note the four spaces between heading and level.  But you can't enter that into the speech dictionary as NVDA compiles it as two things (this is a heading, and this is a level 1 heading).  What you COULD do is create two dictionary entries, and have one which changes "heading" to "h" and another which changes "level 1" to "1".  That does do what you want, but I appreciate the effort required.  It is also something you may want in NVDA's announcement of heading levels, but not when reading those words in an email.  Perhaps a new speech dictionary could be added which affects text in NVDA's strings only?  Feel free to create a specific issue for that..  The ability to save and load "groups" of dictionary entries could potentially be another issue which might make this easier - as someone could do the work and then make it available to others.  Alternatives a "short" and "long" option in say document formatting settings could be used to toggle between things like "heading level" and "h", although it gets more complex with things like superscript and subscript, but if someone wanted to propose an alternative for each, go ahead and create an issue.

**Comment by Jykke67** [comment]
Hi everybody involved in this issue of high priority,

I do fully agree with the arguments that have been presented here, but i do see at least one drawback when using dictionaries for this:

How do we prevent the dictionaries being used when "heading level" appears in a text that we type or read if we just wish to have the announcement of heading level replaced with something else without affecting anything else?

This have been solved in the Earcons and speech rules add-on that seems even to be language undependent; heading level is always replaced with h or whatever the user wishes, and this is more than excellent.

When having said this, the ideal situation would be that the features of the Earcons and speech rules add-on would be implemented to NVDA as configurable options.

In JAWS this has been made by Sound Schemes that can be separately modified. The benefit with this is that modifications of Sound Schemes only affect the announcements, not written or read text.

This is the case even in Earcons and speech rules, which has made navigating on web pages and documents containing headings of various levels much more convenient than before. I have also told this to the author.

**Comment by hozosch** [comment]
Well, the idea was to have a dictionary that only affects the NVDA strings, and nothing you type, from what I was able to get out of Quentin's comment. And yes, Earcons and Speech Rules kind of solves the issue, but currently, "h1", "h2", etc. is spoken additionally to what the user has specified. That was my experience anyway, and it shouldn't be like that.

**Comment by Jykke67** [comment]
Well, and apologies for my missunderstanding. You were right!
